src/components/number-input/number-input.ts

```diff
@@ -51,6 +59,27 @@ export default class BXNumberInput extends BXInput {
     this._input.stepDown();
   }
 
+  _testValidity() {
+    if (this._input?.valueAsNumber > Number(this.max)) {
+      return NUMBER_INPUT_VALIDATION_STATUS.EXCEEDED_MAXIMUM;
+    }
+    if (this._input?.valueAsNumber < Number(this.min)) {
+      return NUMBER_INPUT_VALIDATION_STATUS.EXCEEDED_MINIMUM;
+    }
+    return super._testValidity();
+  }
+
+  _getValidityMessage(state: string) {
+    if (Object.keys(NUMBER_INPUT_VALIDATION_STATUS).includes(state)) {
```

Comment:
```suggestion
    if (Object.values(NUMBER_INPUT_VALIDATION_STATUS).includes(state)) {
```

Seems that `Object.keys(NUMBER_INPUT_VALIDATION_STATUS)` returns `'EXCEEDED_MAXIMUM'` whereas the `state` is `'exceeded_maximum'`.

Reply:
Doh 🤦 yeah that would make sense